core/src/main/java/org/infinispan/util/concurrent/locks/impl/DefaultLockManager.java

```diff
@@ -465,6 +531,10 @@ private <T> T checkState(LockState state, Supplier<T> acquired, Function<Throwab
                }
             }
          }
+
+         if (rv == null)
```

Comment:
This also happens with concurrent ControlLockCommand and the DeadlockProbeCommand. The locks are successfully acquired but before the lock state updates to `ACQUIRED,` another thread changes it. The calls to `.lock()` don't throw any exception and leads to an NPE.